enterprise/cmd/frontend/internal/dotcom/productsubscription/licenses_graphql.go

```diff
@@ -173,6 +178,26 @@ func (r ProductSubscriptionLicensingResolver) ProductLicenses(ctx context.Contex
 	return &productLicenseConnection{db: r.DB, opt: opt}, nil
 }
 
+func (r *ProductSubscriptionLicensingResolver) RevokeLicense(ctx context.Context, args *graphqlbackend.RevokeLicenseArgs) error {
+	// 🚨 SECURITY: Only site admins may revoke product licenses.
+	if err := auth.CheckCurrentUserIsSiteAdmin(ctx, r.DB); err != nil {
+		return err
+	}
+
+	// check if the UUID is valid
+	id, err := uuid.Parse(args.ID)
+	if err != nil {
+		return err
+	}
+
+	err = dbLicenses{db: r.DB}.Revoke(ctx, id.String(), args.Reason)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
can be simplified, right?
```suggestion
	return dbLicenses{db: r.DB}.Revoke(ctx, id.String(), args.Reason)
```

Reply:
Oops, Cody, you can do better than this.

Reply:
haha, yep, all of them still need proofreading from a human being. but maybe not for a long time...